File: sample-chat-obj-c/sample-chat/Components/DialogsScreen/Views/DialogTableViewCell.m
//
//  DialogTableViewCell.m
//  sample-chat
//
//  Created by Injoit on 2/25/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

#import "DialogTableViewCell.h"

@implementation DialogTableViewCell

- (void)awakeFromNib {
    [super awakeFromNib];
    
    self.unreadContainerView.layer.cornerRadius = 10.0f;
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    UIColor *color = self.unreadContainerView.backgroundColor;
    [super setSelected:selected animated:animated];
    
    if (selected) {
        self.unreadContainerView.backgroundColor = color;
    }
}

- (void)setHighlighted:(BOOL)highlighted animated:(BOOL)animated {
    UIColor *color = self.unreadContainerView.backgroundColor;
    [super setHighlighted:highlighted animated:animated];
    
    if (highlighted) {
        self.unreadContainerView.backgroundColor = color;
    }
}

@end

File: sample-chat-swift/sample-chat-swift/Classes/Views/DialogTableViewCell.swift
//
//  DialogTableViewCell.swift
//  sample-chat-swift
//
//  Created by Gleb Ustimenko on 07.07.15.
//  Copyright (c) 2015 quickblox. All rights reserved.
//

import UIKit

class DialogTableViewCell: UITableViewCell {
    
//    @IBOutlet private weak var colorMarker: CornerView!
    
    @IBOutlet weak var dialogLastMessage: UILabel!
    @IBOutlet weak var dialogName: UILabel!
    @IBOutlet weak var dialogTypeImage: UIImageView!
    @IBOutlet weak var unreadMessageCounterLabel: UILabel!
    
    @IBOutlet weak var unreadMessageCounterHolder: UIView!
    
    var dialogID = ""
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        self.unreadMessageCounterHolder.layer.cornerRadius = 10.0
    }
    
    override func setSelected(_ selected: Bool, animated: Bool) {
        
        let markerColor = self.unreadMessageCounterHolder.backgroundColor
        
        super.setSelected(selected, animated: animated)
        
        self.unreadMessageCounterHolder.backgroundColor = markerColor
    }
    
    override func setHighlighted(_ highlighted: Bool, animated: Bool) {
        
        let markerColor = self.unreadMessageCounterHolder.backgroundColor
        
        super.setHighlighted(highlighted, animated: animated)
        
        self.unreadMessageCounterHolder.backgroundColor = markerColor
    }
}
